engine/schema/src/com/cloud/network/dao/NetworkDaoImpl.java

```diff
@@ -377,11 +377,16 @@ protected void addAccountToNetwork(final long networkId, final long accountId, f
     }
 
     @Override
-    public String getNextAvailableMacAddress(final long networkConfigId) {
+    public String getNextAvailableMacAddress(final long networkConfigId, Integer zoneMacIdentifier) {
         final SequenceFetcher fetch = SequenceFetcher.getInstance();
-
         long seq = fetch.getNextSequence(Long.class, _tgMacAddress, networkConfigId);
-        seq = seq | _prefix << 40 | _rand.nextInt(Short.MAX_VALUE) << 16 & 0x00000000ffff0000l;
+        if(zoneMacIdentifier!=0){
```

Comment:
What is `zoneMacIdentifier` is `null`?  Should there be a `Preconditions.checkArgument` check added or a different behavior?